### PR TITLE
3306 Update document of DiceLoss

### LINKED ISSUE #3306: Documentation for DiceLoss is misleading
The [documentation for `DiceLoss`](https://docs.monai.io/en/latest/losses.html?highlight=dice%20loss#diceloss) says (my bold):

> Compute average Dice loss between two tensors. It can support both multi-classes and multi-labels tasks. Input **logits** _input_ (BNHW[D] where N is number of classes) is compared with ground truth _target_ (BNHW[D]). Axis N of _input_ is expected to have **logit** predictions for each class rather than being image channels, while the same axis of target can be 1 or N (one-hot format)...

The term "logit" here doesn't seem to match how `DiceLoss` [actually works](https://github.com/Project-MONAI/MONAI/blob/dev/monai/losses/dice.py#L162-L176); it expects class probabilities and not logit values. That can change for one of the inputs if the parameters `sigmoid` or `softmax` are set to `True`, but those parameters are `False` by default.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #3306 .

### Description
This PR updated the document of `DiceLoss` to make the logits or probabilities input clear.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/losses/dice.py [hunk 1]
```diff
@@ -27,16 +27,16 @@
...
-    values added to the intersection and union components of the inter-over-union calculation to smooth results
-    respectively, these values should be small. The `include_background` class attribute can be set to False for
-    an instance of DiceLoss to exclude the first category (channel index 0) which is by convention assumed to be
-    background. If the non-background segmentations are small compared to the total image size they can get
-    overwhelmed by the signal from the background so excluding it in such cases helps convergence.
+    The data `input` (BNHW[D] where N is number of classes) is compared with ground truth `target` (BNHW[D]).
 
-    Milletari, F. et. al. (2016) V-Net: Fully Convolutional Neural Networks forVolumetric Medical Image Segmentation, 3DV, 2016.
+    Note that axis N of `input` is expected to be logits or probabilities for each class, must set `sigmoid=True`
+    or `softmax=True` if passing logits as input. And the same axis of `target` can be 1 or N (one-hot format).
```
Comment (yiheng-wang-nv) [inline monai/losses/dice.py:33]: or specifying `other_act`

Comment (Nic-Ma) [reply monai/losses/dice.py]: Sure, updated in the latest commit.

Thanks.

**Review by yiheng-wang-nv — APPROVED** [review]
